src/main/java/fi/metatavu/dcfb/server/rest/AbstractApi.java

```diff
@@ -128,30 +129,58 @@ protected String slugify(String text) {
   }
 
   /**
-   * Parses CSV enum parameter from string list into enum list
+   * Returns list parameter as <E> translated by given translate function.
    * 
-   * @param enumType target enum class
-   * @param parameters string values
-   * @return list of enums
-   * @throws IllegalArgumentException if parameters contain invalid values
+   * @param parameter list parameter as string list
+   * @param translate translate function
+   * @return list of <E>
    */
-  @SuppressWarnings ("squid:S1168")
-  protected <T extends Enum<T>> List<T> getEnumListParameter(Class<T> enumType, List<String> parameter) {
+  protected <E> List<E> getListParameter(List<String> parameter, Function<String, E> translate) {
     if (parameter == null) {
       return null;
     }
-    
-    List<String> merged = new ArrayList<>(parameter.size());
+
+    List<String> merged = new ArrayList<>();
     
     parameter.stream()
       .filter(StringUtils::isNoneEmpty)
       .forEach(filter -> merged.addAll(Arrays.asList(StringUtils.split(filter, ','))));
 
     return merged.stream()
-      .map(name -> Enum.valueOf(enumType, name))
+      .map((name) -> {
+        return translate.apply(name);
+      })
       .collect(Collectors.toList());
   }
 
+  /**
+   * Returns list parameter as <E> translated by given translate function.
+   * 
+   * @param parameter list parameter as string
+   * @param translate translate function
+   * @return list of <E>
+   */
+  protected <E> List<E> getListParameter(String parameter, Function<String, E> translate) {
+    if (parameter == null) {
+      return null;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Return an empty collection instead of null. ([squid:S1168](https://sonarcloud.io/organizations/metatavu-oy/rules#rule_key=squid%3AS1168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fi.metatavu.dcfb-api%3Adcfb-api&pullRequest=11&issues=AWQt_bhuYwZU9vvrs3Ky&open=AWQt_bhuYwZU9vvrs3Ky)

[ISSUE_KEY]: AWQt_bhuYwZU9vvrs3Ky